pytroll-exercise-2018-10/5-custom-reader.ipynb

```diff
@@ -0,0 +1,383 @@
+    "\n",
+    "### The `reader` section\n",
+    "The `reader` section, that gives a general parameters of the reader.\n",
+    "\n",
+    "The parameters to provide in this section are:\n",
+    "- description: General description of the reader\n",
+    "- name: this is the name of the reader, it should be the same as the filename (without the .yaml extension obviously).  This is the name used interactively in satpy, so choose it well! A loose convention is to use `<format>_<instrument>_<level>` as a template for the name\n",
+    "- sensors: the list of sensors this reader will support\n",
+    "- reader: the metareader to use, in most cases the `FileYAMLReader` is a good choice."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```yaml\n",
+    "reader:\n",
+    "  description: NetCDF4 reader for the Eumetsat MSG format\n",
+    "  name: nc_seviri_l1b\n",
+    "  sensors: [seviri]\n",
+    "  reader: !!python/name:satpy.readers.yaml_reader.FileYAMLReader\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### The `file_types` section\n",
+    "\n",
+    "Each file type needs to provide:\n",
+    " - `file_reader`, the class that will handle the files for this reader, that you will implement in the corresponding python file (see next section)\n",
+    " - `file_patterns`, the patterns to match to find files this reader can handle. The syntax to use is basically the same as `format` with the addition of time. See the [trollsift package documentation](https://trollsift.readthedocs.io/en/latest/usage.html) for more details.\n",
+    " - Optionally, a file type can have a `requires` field: it is a list of file types that the current file types needs to function. For example, the HRIT MSG format segment files each need a prologue and epilogue file to be read properly, hence in this case we have added `requires: [HRIT_PRO, HRIT_EPI]` to the file type definition.\n",
+    " \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```yaml\n",
+    "file_types:\n",
+    "    nc_seviri_l1b:\n",
+    "        file_reader: !!python/name:satpy.readers.nc_seviri_l1b.NCSEVIRIFileHandler\n",
+    "        file_patterns: ['W_XX-EUMETSAT-Darmstadt,VIS+IR+IMAGERY,{satid:4s}+SEVIRI_C_EUMG_{processing_time:%Y%m%d%H%M%S}.nc']\n",
+    "        nc_seviri_l1b_hrv:\n",
+    "        file_reader: !!python/name:satpy.readers.nc_seviri_l1b.NCSEVIRIHRVFileHandler\n",
+    "        file_patterns: ['W_XX-EUMETSAT-Darmstadt,HRV+IMAGERY,{satid:4s}+SEVIRI_C_EUMG_{processing_time:%Y%m%d%H%M%S}.nc']\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### The `datasets` section\n",
+    "\n",
+    "The datasets section describes each dataset available in the files. The parameters provided are made available to the methods of the implementing class.\n",
```

Comment:
"made available to the methods of the implementing class" didn't make sense to me at first, but I suppose this is good enough. Or maybe "These parameters are made available to the file handling python code." or something like that.